CsCore/PlainNetClassLib/src/Plugins/CsCore/com/csutil/http/apis/OpenAi.cs

```diff
@@ -170,40 +159,86 @@ public class Request
 
             }
 
-            public class Response
-            {
+            public class Response {
                 public int created { get; set; }
                 public List<ImageEntry> data { get; set; }
 
-                public class ImageEntry
-                {
+                public class ImageEntry {
                     public string url { get; set; }
                 }
 
             }
 
         }
 
+        /// <summary> https://platform.openai.com/docs/guides/text-to-speech/text-to-speech
+        /// </summary>
+        public async Task<Stream> TextToSpeech(Audio.TTSRequest requestParam) {
+            return await new Uri("https://api.openai.com/v1/audio/speech").SendPOST().WithAuthorization(apiKey)
+            .WithJsonContent(requestParam).GetResult<Stream>();
+        }
+
+        /// <summary> https://platform.openai.com/docs/guides/speech-to-text
+        /// </summary>
+        public async Task<Audio.STTResponse> SpeechToText(Audio.STTRequest requestParam) {
+            Dictionary<string, object> formContent = new Dictionary<string, object>();
+            formContent.Add("model", requestParam.model);
+            formContent.Add("language", requestParam.language);
+            formContent.Add("responseFormat", requestParam.responseFormat);
+            formContent.Add("temperature", requestParam.temperature);
+            if (requestParam.prompt != null) {
+                formContent.Add("prompt", requestParam.prompt);
+            }
+
+            DirectoryEntry dir = EnvironmentV2.instance.GetNewInMemorySystem();
+            FileEntry fileToUpload = dir.GetChild("speech.mp3");
+            await fileToUpload.SaveStreamAsync(requestParam.fileStream, resetStreamToStart: false);
```

Comment:
Check how .AddFileViaForm is implemented to skip the entire  fileToUpload.SaveStreamAsync 
I am sure you can upload the fileStream directly as well

Reply:
we tried adding stream directly to formContent(dictionary), but it for some reason does not work. `Dictionary<string, object> formContent = new Dictionary<string, object>();
var streamContent = new StreamContent(requestParam.fileStream);
            streamContent.Headers.ContentDisposition = new System.Net.Http.Headers.ContentDispositionHeaderValue("form-data") {
                Name = "file",
                FileName = "file"
            };`